device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -276,12 +276,14 @@ else if (message != null)
 
         try
         {
-            String messageId = message.getCorrelationId();
-
-            if (messageId != null && correlationCallbacks.containsKey(messageId))
+            if (message != null)
```

Comment:
Detected a null check was required here for messageId

Reply:
You may want to add an else case that simply logs a warning that "A message was received with no correlation Id"